pyiron_workflow/snippets/files.py

```diff
@@ -145,3 +145,33 @@ def _resolve_directory_and_path(
         if not isinstance(directory, DirectoryObject):
             directory = DirectoryObject(directory)
         return file_name, directory
+
+    def copy(
+        self,
+        new_file_name: str | None=None,
+        directory: DirectoryObject | str | None=None
```

Comment:
Use `Optional` and I guess the `directory` hint needs to be expanded to include `Path` now?

Reply:
No currently it doesn't include `Path`. It's a huge mess, I agree XD

Reply:
> According to [what I see in the Internet](https://stackoverflow.com/questions/69440494/python-3-10-optionaltype-or-type-none), type hinting with `|` is preferred over `Optional`.

tldr; ok, then do whatever you want.

I'm actually quite ambivalent about this. On the one hand, I appreciate not needing to import `Optional`, on the other hand I feel that `foo: Optional[int] = None` is preferable because (a) it doesn't have `None` repeated, and (b) my brain can register that this is, in fact, optional only two characters after the `:`. In contrast, `foo: int | None = None` has ugly repetition and I can't use early stopping in my reading because I don't find out it's optional until the end. This is why ["There should be one-- and preferably only one --obvious way to do it."](https://peps.python.org/pep-0020/) 😠 😂

Reply:
I agree that `Optional[str] = None` looks more elegant than `str | None = None`, but since there's too much work to do to care about elegance I'm gonna merge this one XD